addons/analytic/models/analytic.py

```diff
@@ -19,27 +19,29 @@ class account_analytic_account(models.Model):
 
     @api.multi
     def _compute_debit_credit_balance(self):
-        analytic_line_obj = self.env['account.analytic.line']
-        domain = [('account_id', 'in', self.mapped('id'))]
+        sql_query = """
+            SELECT account_id, SUM(amount)
+            FROM account_analytic_line
+            WHERE account_id IN %s
+            AND company_id = %s
+        """
+        sql_params = (tuple(self.ids),
+                      self.env.user.company_id.id)
         if self._context.get('from_date', False):
-            domain.append(('date', '>=', self._context['from_date']))
+            sql_query += " AND date >= %s"
+            sql_params += (self._context['from_date'],)
         if self._context.get('to_date', False):
-            domain.append(('date', '<=', self._context['to_date']))
-
-        account_amounts = analytic_line_obj.search_read(domain, ['account_id', 'amount'])
-        account_ids = set([line['account_id'][0] for line in account_amounts])
-        data_debit = {account_id: 0.0 for account_id in account_ids}
-        data_credit = {account_id: 0.0 for account_id in account_ids}
-        for account_amount in account_amounts:
-            if account_amount['amount'] < 0.0:
-                data_debit[account_amount['account_id'][0]] += account_amount['amount']
-            else:
-                data_credit[account_amount['account_id'][0]] += account_amount['amount']
+            sql_query += " AND date <= %s"
+            sql_params += (self._context['to_date'],)
```

Comment:
sql_params should probably be a list, and then these would just be `sql_params.append(…)`.